harvester_e2e_tests/integration/test_rancher_integration.py

```diff
@@ -0,0 +1,319 @@
+class TestRKE2:
+    def test_create_mgmt_cluster(self, rancher_api_client, unique_name, wait_timeout):
+        code, data = rancher_api_client.mgmt_clusters.create(
+            self.get_harvester_cluster_name(unique_name),
+            labels={
+                "provider.cattle.io": "harvester"
+            }
+        )
+
+        assert 201 == code, (
+            f"Failed to create MgmtCluster {self.get_harvester_cluster_name(unique_name)} with \
+                error: {code}, {data}"
+        )
+
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        while endtime > datetime.now():
+            code, data = rancher_api_client.mgmt_clusters.get(
+                self.get_harvester_cluster_name(unique_name))
+            if 'status' in data and 'clusterName' in data['status']:
+                break
+            sleep(5)
+        else:
+            raise AssertionError(
+                f"Can't find clusterName in MgmtCluster \
+                    {self.get_harvester_cluster_name(unique_name)} with {wait_timeout} timed out\n"
+                f"Got error: {code}, {data}"
+            )
+
+    def test_import_harvester_to_rancher(self, api_client, rancher_api_client, unique_name,
+                                         wait_timeout):
+        code, data = rancher_api_client.mgmt_clusters.get(
+            self.get_harvester_cluster_name(unique_name)
+        )
+        clusterName = data['status']['clusterName']
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        while endtime > datetime.now():
+            code, data = rancher_api_client.cluster_registration_tokens.get(clusterName)
+            if code == 200 and 'manifestUrl' in data:
+                break
+            sleep(5)
+        else:
+            raise AssertionError(
+                f"Can't find clusterRegistrationToken for cluster {clusterName} \
+                    with {wait_timeout} timed out\n"
+                f"Got error: {code}, {data}"
+            )
+
+        manifestUrl = data['manifestUrl']
+        updates = {
+            "value": manifestUrl
+        }
+        code, data = api_client.settings.update("cluster-registration-url", updates)
+        assert 200 == code, (
+            f"Failed to update cluster-registration-url setting with error: {code}, {data}"
+        )
+
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        while endtime > datetime.now():
+            code, data = rancher_api_client.mgmt_clusters.get(
+                self.get_harvester_cluster_name(unique_name)
+            )
+            if 'status' in data and 'ready' in data['status'] and data['status']['ready']:
+                break
+            sleep(5)
+        else:
+            raise AssertionError(
+                f"MgmtCluster {self.get_harvester_cluster_name(unique_name)} can't be ready \
+                    with {wait_timeout} timed out\n"
+                f"Got error: {code}, {data}"
+            )
```

Comment:
need to add the check point to make sure the cluster imported and **State** is `Active` in rancher.